edx_exams/apps/api/v1/views.py

```diff
@@ -179,3 +179,78 @@ def patch(self, request, course_id):
             data = {"detail": "Invalid data", "errors": serializer.errors}
 
         return Response(status=response_status, data=data)
+
+
+class CourseExamConfigurationsView(APIView):
+    # todo: make parameters descriptions more useful.
+    """
+    View to create and update course exam configs for a specific course.
+
+    Given a course id and a proctoring provider, this view will either create a new course exam configuration
+    (if one doesn't exist), or modify the proctoring provider on an existing course exam config.
+
+    HTTP PATCH
+    Creates or updates a CourseExamConfiguration.
+    Expected PATCH data: {
+        'name': 'test_provider',
+        'verbose_name': 'testing_provider',
+        'lti_configuration_id': '123456789'
+    }
+    **PATCH data Parameters**
+        * name: This is the name of the proctoring provider.
+        * verbose_name: This is the verbose name of the proctoring provider.
+        * lti_configuration_id: This is the LTI configuration id of the proctoring provider.
+    **Exceptions**
+        * HTTP_400_BAD_REQUEST
+    """
+
+    authentication_classes = (JwtAuthentication,)
+    permission_classes = (StaffUserPermissions,)
+
+    @classmethod
+    def handle_config(cls, provider, course_id):
+        """
+        Helper method that decides whether to update existing or create new config.
+        """
+        # todo: is there a better way to check whether the config already exists?
+        # todo: does the course_id need a uniqueness constraint in the CourseExamConfiguration model?
+        config = CourseExamConfiguration.objects.filter(course_id=course_id)
+
+        # update if config exists
+        if len(config) == 1:
+            config[0].provider = provider
+            config[0].save()
+            log.info("Updated course exam configuration provider=%(provider_name)s",
+                     {
+                         'provider_name': provider.name,
+                     }
+                     )
+        # else create new config
+        else:
+            config = CourseExamConfiguration.objects.create(course_id=course_id, provider=provider)
+            config.save()
+            log.info("Created course exam configuration course_id=%(course_id)s, provider=%(provider_name)s",
+                     {
+                         'course_id': config.course_id,
+                         'provider_name': provider.name,
+                     }
+                     )
+
+    def patch(self, request, course_id):
+        """
+        Create/update course exam configuration.
+        """
+
+        serializer = ProctoringProviderSerializer(data=request.data)
+
+        if serializer.is_valid():
+            provider = serializer.save()
+            self.handle_config(provider, course_id)
```

Comment:
Part of this ticket is to determine if we also need to update a course's exams after the config is changed. I do think we need that behavior, so it should be handled as part of this view.

Reply:
Yes, you're right @alangsto! So @Zacharis278 had mentioned maybe needing to wait on this until [this decision was made](https://2u-internal.atlassian.net/wiki/spaces/PT/pages/95223955/Discovery+Altering+Active+Exams). In the meantime, do you think it's best to move forward as is for this ticket then create a new ticket to add this behavior?

Reply:
I've created that ticket [here](https://2u-internal.atlassian.net/browse/MST-1579).

Let me know if anything's missing in that ticket. :)